### PR TITLE
GEODE-3539: refactor CreateIndexCommand and add more tests

### PR DESCRIPTION
**Pull request by jinmeiliao**
This is part of the effort to add more test coverage of commands

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/test/junit/rules/GfshParserRule.java [hunk 1]
```diff
@@ -14,8 +14,18 @@
  */
 package org.apache.geode.test.junit.rules;
 
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (pdxrunner) [inline geode-core/src/test/java/org/apache/geode/test/junit/rules/GfshParserRule.java]: Remove unused import.

**Comment by jaredjstewart** [comment]
We probably want to a unit test to validate the behavior of the new converter.

File: geode-web/src/test/java/org/apache/geode/management/internal/cli/commands/IndexCommandOverHttpTest.java [hunk 2]
```diff
@@ -17,14 +17,15 @@
 
 import org.junit.experimental.categories.Category;
 
+import org.apache.geode.test.junit.categories.IntegrationTest;
 import org.apache.geode.test.junit.rules.GfshShellConnectionRule;
 import org.apache.geode.test.dunit.rules.MemberVM;
```
Comment (pdxrunner) [inline geode-web/src/test/java/org/apache/geode/management/internal/cli/commands/IndexCommandOverHttpTest.java]: MemberVM is now an unused import

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateIndexCommand.java [hunk 3]
```diff
@@ -78,118 +77,94 @@ public Result createIndex(@CliOption(key = CliStrings.CREATE_INDEX__NAME, mandat
...
-      } else {
-        return ResultBuilder
-            .createUserErrorResult(CliStrings.CREATE_INDEX__INVALID__INDEX__TYPE__MESSAGE);
-      }
-
-      if (indexName == null || indexName.isEmpty()) {
-        return ResultBuilder.createUserErrorResult(CliStrings.CREATE_INDEX__INVALID__INDEX__NAME);
-      }
-
-      if (indexedExpression == null || indexedExpression.isEmpty()) {
```
Comment (jaredjstewart) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateIndexCommand.java]: Why is it safe to remove these validation check for `indexName` and `indexedExpression`?

Comment (jinmeiliao) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateIndexCommand.java]: these are marked as "mandatory", so the parser should have already done the validation.  See the newly added CreateIndexCommandTest that covers these validations. 

Rule of thumb is: at the point (command exec body), we should have been beyond individual option validation already. This should only need to focus on the execution of the command. Validation is better done by the parser, the converter or the interceptor.

Comment (jaredjstewart) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateIndexCommand.java]: Ah, I couldn't see the "mandatory" requirement in the collapsed diff. Looks good to me!

**Pull request merged**